public/docs/_examples/server-communication/dart/web/main.dart

```diff
@@ -0,0 +1,25 @@
+// #docregion
+import 'package:angular2/angular2.dart';
+import 'package:angular2/bootstrap.dart';
+import 'package:http/browser_client.dart';
+import 'package:http_in_memory_web_api/http_in_memory_web_api.dart';
+import 'package:server_communication/toh/toh_component.dart';
+import 'package:server_communication/hero_data.dart';
+import 'package:server_communication/wiki/wiki_component.dart';
+import 'package:server_communication/wiki/wiki_smart_component.dart';
+
+main() {
+  bootstrap(TohComponent, [
+//#enddocregion
+//#docregion in-mem-web-api-providers
+// in-memory web api providers
+    provide(BrowserClient,
+        useValue:
+            new HttpClientInMemoryBackendService(heroData)), // in-mem server
```

Comment:
@thso @kwalrath The only way I could make it work was like this:

```dart
import 'package:angular2/angular2.dart';
import 'package:server_communication/hero_data.dart';
import 'package:http/browser_client.dart';
import 'package:http_in_memory_web_api/http_in_memory_web_api.dart';
import 'hero_service.dart';
import 'hero_list_component.dart';

seedData() => heroData();

newBrowserClientProvider(heroData) =>
    new HttpClientInMemoryBackendService(heroData);

@Component(
    selector: 'my-toh',
// #docregion template
    template: '''
      <h1>Tour of Heroes</h1>
      <hero-list></hero-list>
    ''',
// #enddocregion template
    providers: const [
      const Provider(CreateDb, useValue: seedData),
      const Provider(BrowserClient,
          useFactory: newBrowserClientProvider, deps: const [CreateDb]),
      HeroService
    ],
    directives: const [
      HeroListComponent
    ])
class TohComponent {}
// #enddocregion

```

When I do this, seedData or newBrowserClientProvider cannot be private otherwise I get this error:
> main.dart:1 Internal error: 'package:server_communication/toh/toh_component.template.dart': error: line 28 pos 136: expression is not a valid compile-time constant
const [const Component(directives: const [HeroListComponent], providers: const [HeroService, const Provider(BrowserClient, useFactory: _newBrowserClientProvider)], selector: 'my-toh', template: '''

I also could not provide `CreateDb` with `useFactory`, I also got some reflection errors.
```
newCreateDBProvider() => heroData;

...

const Provider(CreateDb, useFactory: newCreateDBProvider)
```
> EXCEPTION: Cannot find reflection information on newCreateDBProvider

Do you still think it is worth to keep it here instead of the `bootstrap` ?